src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils.ts

```diff
@@ -35,15 +37,23 @@ export const getAccountAddresses = memoize(async (accountId: AccountId): Promise
 
     if (!addresses) return []
 
-    return addresses.map(address =>
-      address.pubkey.startsWith('bitcoincash')
-        ? address.pubkey.replace('bitcoincash:', '')
-        : address.pubkey,
-    )
+    return addresses.map(({ pubkey, balance }) => {
+      const address = pubkey.startsWith('bitcoincash') ? pubkey.replace('bitcoincash:', '') : pubkey
+
+      return { address, balance }
+    })
   }
 
-  return [fromAccountId(accountId).account]
-})
+  // We don't need balances for chain others than UTXOs
+  return [{ address: fromAccountId(accountId).account, balance: '' }]
+}
+
+// Memoized on accountId, see lodash docs:
+// "By default, the first argument provided to the memoized function is used as the map cache key."
+export const getAccountAddresses = memoize(
```

Comment:
be careful memoizing here - remember after every UTXO tx we'll have new addresses.

Reply:
AKCHYUALLY fine